plugins/modules/ec2_vpc_vgw.py

```diff
@@ -270,50 +291,48 @@ def create_vgw(client, module):
         params["AmazonSideAsn"] = module.params.get("asn")
 
     try:
-        response = client.create_vpn_gateway(aws_retry=True, **params)
-        get_waiter(client, "vpn_gateway_exists").wait(VpnGatewayIds=[response["VpnGateway"]["VpnGatewayId"]])
+        response = create_vpn_gateway(client, **params)
+        get_waiter(client, "vpn_gateway_exists").wait(VpnGatewayIds=[response["VpnGatewayId"]])
     except botocore.exceptions.WaiterError as e:
-        module.fail_json_aws(
-            e, msg=f"Failed to wait for Vpn Gateway {response['VpnGateway']['VpnGatewayId']} to be available"
-        )
+        module.fail_json_aws(e, msg=f"Failed to wait for Vpn Gateway {response['VpnGatewayId']} to be available")
     except is_boto3_error_code("VpnGatewayLimitExceeded") as e:
         module.fail_json_aws(e, msg="Too many VPN gateways exist in this account.")
-    except (
-        botocore.exceptions.ClientError,
-        botocore.exceptions.BotoCoreError,
-    ) as e:  # pylint: disable=duplicate-except
-        module.fail_json_aws(e, msg="Failed to create gateway")
+    except AnsibleEC2Error as e:
```

Comment:
Should this be handled here? Isn't this covered by https://github.com/ansible-collections/amazon.aws/pull/2331/files#diff-11d6a927433f45aa84de2f54fa0adf33e8094ebb417c75ea476ffe61b4e4f587R652 ?

Reply:
removed.